src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SchemaUtils.java

```diff
@@ -0,0 +1,539 @@
+      databaseClient
+        .singleUse()
+        .executeQuery(statementBuilder.build());
+
+    Set<String> result = new HashSet<>();
+    while (resultSet.next()) {
+      result.add(resultSet.getString(INFORMATION_SCHEMA_TABLE_NAME));
+    }
+    resultSet.close();
+
+    return result;
+  }
+
+  /**
+   * @return if the Change Stream is tracking all the tables in the database.
+   */
+  public static boolean isChangeStreamForAll(DatabaseClient databaseClient,
+                                               String changeStreamName) {
+    final String sql =
+      "SELECT CHANGE_STREAMS.ALL FROM INFORMATION_SCHEMA.CHANGE_STREAMS " +
+        "WHERE CHANGE_STREAM_NAME = @changeStreamName";
+    final ResultSet resultSet =
+      databaseClient
+        .singleUse()
+        .executeQuery(
+          Statement.newBuilder(sql)
+            .bind("changeStreamName").to(changeStreamName)
+            .build());
+
+    Boolean result = null;
+    while (resultSet.next()) {
+      if (resultSet.getBoolean(INFORMATION_SCHEMA_ALL)) {
+        result = true;
+      } else {
+        result = false;
+      }
+    }
+    resultSet.close();
+
+    if (result == null) {
+      throw new IllegalArgumentException(
+        String.format("Cannot find change stream %s in INFORMATION_SCHEMA", changeStreamName));
+    }
+
+    return result;
+  }
+
+  /**
+   * @return the Spanner column names that are tracked explicitly by Change Streams by table name.
+   * e.g. Given the following table:
+   * CREATE TABLE Singers (
+   * SingerId   INT64 NOT NULL,
+   * FirstName  STRING(1024),
+   * LastName   STRING(1024),
+   * );
+   * <p>
+   * Return an empty map if we have the following Change Streams:
+   * CREATE CHANGE STREAM AllStream FOR ALL
+   * <p>
+   * Return {"Singers" -> {"SingerId", "LastName"}} if we have the following Change Streams:
+   * CREATE CHANGE STREAM SingerStream FOR Singers(SingerId, FirstName)
+   */
+  private static Map<String, Set<String>>
+  getSpannerColumnNamesExplicitlyTrackedByChangeStreamsByTableName(
+    DatabaseClient databaseClient, String changeStreamsName) {
+    final String sql =
+      "SELECT TABLE_NAME, COLUMN_NAME FROM INFORMATION_SCHEMA.CHANGE_STREAM_COLUMNS "
+        + "WHERE CHANGE_STREAM_NAME = @changeStreamName";
+
+    final ResultSet resultSet =
+      databaseClient
+        .singleUse()
+        .executeQuery(
+          Statement.newBuilder(sql)
+            .bind("changeStreamName").to(changeStreamsName)
+            .build());
+
+    Map<String, Set<String>> result = new HashMap<>();
+    while (resultSet.next()) {
+      final String tableName = resultSet.getString(INFORMATION_SCHEMA_TABLE_NAME);
+      final String columnName = resultSet.getString(INFORMATION_SCHEMA_COLUMN_NAME);
+      result.putIfAbsent(tableName, new HashSet<>());
+      result.get(tableName).add(columnName);
+    }
+
+    return result;
+  }
+
+  private static Type informationSchemaTypeToSpannerType(String type) {
```

Comment:
String / Type parsing could be its own class

Reply:
I moved this into SpannerUtils, let me know if you still think it should be in its own class.